src/main/resources/config/liquibase/changelog/20230601154000_changelog.xml

```diff
@@ -0,0 +1,128 @@
+                   xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+                   xsi:schemaLocation="http://www.liquibase.org/xml/ns/dbchangelog http://www.liquibase.org/xml/ns/dbchangelog/dbchangelog-4.18.xsd">
+    <changeSet author="baboci" id="20230601154000">
+        <comment>Create new channels for existing lecture/exercise and migrate posts of lecture/exercise to respective channel</comment>
+        <comment>Step 1: Create channels for lectures that do not have a channel and update posts</comment>
+        <sql>
+            INSERT INTO conversation (course_id, lecture_id, name, description,discriminator, is_public, is_announcement, creation_date)
+            SELECT l.course_id, l.id, CONCAT('lecture-',LOWER(REPLACE(LEFT(l.title, 10), ' ', '-')), '-', l.id), CONCAT('Description of lecture: ', l.title), 'C', true, false, NOW()
+            FROM lecture l
+                     LEFT JOIN conversation c ON l.id = c.lecture_id
+                     LEFT JOIN course c2 on l.course_id = c2.id
+            WHERE c.lecture_id IS NULL
+              AND c2.info_sharing_config = 1
+              AND (c2.end_date IS NULL OR c2.end_date > now())
+              AND NOT EXISTS (
+                SELECT 1
+                FROM conversation
+                WHERE name = CONCAT('lecture-', LOWER(REPLACE(LEFT(l.title, 10), ' ', '-')), '-', l.id)
+                );
+        </sql>
+        <sql>
+            UPDATE post p
+            SET conversation_id = (
+                SELECT DISTINCT c.id as conversation_id
+                FROM conversation c
+                         LEFT JOIN lecture l ON l.id = c.lecture_id
+                WHERE p.lecture_id = l.id
+            )
+            WHERE p.lecture_id IS NOT NULL
+              AND p.conversation_id IS NULL;
+        </sql>
+        <sql>
+            UPDATE post
+            SET lecture_id = NULL
+            WHERE conversation_id IS NOT NULL
+              AND lecture_id IS NOT NULL;
+        </sql>
+        <comment>Step 2: Create channels for exercises that do not have a channel and update posts</comment>
+        <sql>
+            INSERT INTO conversation (course_id, exercise_id, name, description,discriminator, is_public, is_announcement, creation_date)
+            SELECT e.course_id, e.id, CONCAT('exercise-', LOWER(REPLACE(LEFT(e.title, 10), ' ', '-')), '-', e.id), CONCAT('Description of exercise: ', e.title), 'C', true, false, NOW()
+            FROM exercise e
+                     LEFT JOIN conversation c ON e.id = c.exercise_id
+                     LEFT JOIN course c2 on e.course_id = c2.id
+            WHERE c.exercise_id IS NULL
+              AND c2.info_sharing_config = 1
+              AND (c2.end_date IS NULL OR c2.end_date > now())
+              AND NOT EXISTS (
+                SELECT 1
+                FROM conversation
+                WHERE name = CONCAT('lecture-', LOWER(REPLACE(LEFT(e.title, 10), ' ', '-')), '-', e.id)
```

Comment:
why don't use the e.id = conversation.exercise_id condition here to check if an exercise already has a channel or not?

Reply:
This check is here to make sure that the create channel does not conflict with any other channel name within the course. So if there is channel with same name the new channel will not be created.